XCUITests/TopTabsTest.swift

```diff
@@ -0,0 +1,162 @@
+        let tabsOpen = app.buttons["Show Tabs"].value
+        XCTAssertEqual("2", tabsOpen as? String)
+
+        // The tab tray shows the correct tabs
+        navigator.goto(TabTray)
+        waitforExistence(app.collectionViews.cells[urlLabel])
+    }
+
+    func testAddTabFromContext() {
+        navigator.openURL(urlString: urlExample)
+        // Initially there is only one tab open
+        let tabsOpenInitially = app.buttons["Show Tabs"].value
+        XCTAssertEqual("1", tabsOpenInitially as? String)
+
+        // Open link in a different tab and switch to it
+        waitforExistence(app.webViews.links.staticTexts["More information..."])
+        app.webViews.links.staticTexts["More information..."].press(forDuration: 5)
+        app.buttons["Open in New Tab"].tap()
+        app.buttons["Switch"].tap()
+
+        // Open tab tray to check that both tabs are there
+        navigator.goto(TabTray)
+        waitforExistence(app.collectionViews.cells["Example Domain"])
+        waitforExistence(app.collectionViews.cells["IANA — IANA-managed Reserved Domains"])
+        let numTabsOpen = app.collectionViews.cells.count
+        XCTAssertEqual(numTabsOpen, 2, "Tab not added from link")
+    }
+
+    func testSwitchBetweenTabs() {
+        // Open two urls from tab tray and switch between them
+        navigator.openURL(urlString: url)
+        navigator.goto(TabTray)
+        navigator.openURL(urlString: urlYah)
+        navigator.goto(TabTray)
+        waitforExistence(app.collectionViews.cells[urlLabel])
+        app.collectionViews.cells[urlLabel].tap()
+        waitForValueContains(app.textFields["url"], value: urlValue)
+
+        navigator.nowAt(BrowserTab)
+        navigator.goto(TabTray)
+        waitforExistence(app.collectionViews.cells[urlLabelYah])
+        app.collectionViews.cells[urlLabelYah].tap()
+        waitForValueContains(app.textFields["url"], value: urlValueYah)
+    }
+
+    func testCloseOneTab() {
+        navigator.openURL(urlString: url)
+        navigator.goto(NewTabScreen)
+        navigator.goto(TabTray)
+        waitforExistence(app.collectionViews.cells[urlLabel])
+
+        // 'x' button to close the tab is not visible, so closing by swiping the tab
+        app.collectionViews.cells[urlLabel].swipeRight()
+
+        // After removing only one tab is visible and should be home
+        let numTabsOpen = app.collectionViews.cells.count
+        XCTAssertEqual(numTabsOpen, 1, "After removing one tab there should remain only one")
+        waitforNoExistence(app.collectionViews.cells[urlLabel])
+        waitforExistence(app.collectionViews.cells["home"])
+
+    }
+
+    private func openSeveralTabs (numberTabs: Int) {
+        for _ in 1...numberTabs {
+            navigator.goto(NewTabScreen)
+            navigator.goto(TabTray)
+        }
+    }
+
+    private func closeAllTabs() {
+        navigator.goto(TabTrayMenu)
+        app.collectionViews.cells["CloseAllTabsMenuItem"].tap()
```

Comment:
actually, I might have missed the prior conversation on this topic, but tapping 'close all tabs' to go to NewTabScreen in fxscreengraph is kinda odd, because it doesn't hint that all existing tabs will be removed.  It is one way to go to a new tab screen among other methods for sure, but I think the script is much more readable if the closeAllTabs() contains explicit action to close all tabs. I prefer the way it is now.